documentation/concepts/device_scheduler.rst

```diff
@@ -0,0 +1,151 @@
+:math:`P^{ems}(j)`                    ems_power                                          Aggregated power of all the devices at time :math:`j`.
+:math:`Commitment(c,j)`               commitment_quantity                                Commitments made by the EMS.
+================================ ================================================ ==============================================================================================================  
+
+Cost Function
+--------------
+
+The cost function quantifies the total cost of upwards and downwards deviations from the different commitments.
+
+.. math:: 
+    :name: cost_function
+
+    \min [\sum_{c,j} \Delta _{up}(c,j) \cdot Price_{up}(c,j) +  \Delta_{down}(c,j) \cdot Price_{down}(c,j)]
+
+
+State Dynamics
+---------------
+
+To simplify the description of the model, the auxiliary variable :math:`\Delta Stock(d,j)` is introduced in the documentation. It represents the
+change of :math:`Stock(d,j)`, taking into account conversion efficiencies but not considering the storage losses.
+
+.. math::
+  :name: stock
+
+    \Delta Stock(d,j) = \frac{P_{down}(d,j)}{\eta_{down}(d,j) } + P_{up}(d,j)  \cdot \eta_{up}(d,j)
+
+
+
+.. math:: 
+  :name: device_bounds
+
+    Stock_{min}(d,j)  \leq Stock(d,j) - Stock(d,-1)\leq Stock_{max}(d,j) 
+
+
+Perfect Efficiency
+^^^^^^^^^^^^^^^^^^^
+
+.. math:: 
+  :name: efficiency_e1
+
+    Stock(d, j) = Stock(d, j-1) + \Delta Stock(d,j)
+
+Left Efficiency
+^^^^^^^^^^^^^^^^^
+First apply the stock change, then apply the losses (i.e. the stock changes on the left side of the time interval in which the losses apply)
+
+
+.. math:: 
+  :name: efficiency_left
+
+    Stock(d, j)  = (Stock(d, j-1) + \Delta Stock(d,j)) \cdot \epsilon(d,j)
+
+
+Right Efficiency
+^^^^^^^^^^^^^^^^^
+First apply the losses, then apply the stock change (i.e. the stock changes on the right side of the time interval in which the losses apply)
+
+.. math:: 
+  :name: efficiency_right
+
+    Stock(d, j)  = Stock(d, j-1) \cdot \epsilon(d,j) + \Delta Stock(d,j)
+
+Linear Efficiency
+^^^^^^^^^^^^^^^^^
+Assume the change happens at a constant rate, leading to a linear stock change, and exponential decay, within the current interval
+
+.. math:: 
+  :name: efficiency_linear
+
+    Stock(d, j)  = Stock(d, j-1) \cdot \epsilon(d,j) + \Delta Stock(d,j) \cdot \frac{\epsilon(d,j) - 1}{log(\epsilon(d,j))}
+
+Constraints
+--------------
+
+Device bounds
+^^^^^^^^^^^^^
+
+.. math:: 
+  :name: device_derivative_bounds
+
+    P_{min}(d,j) \leq P_{up}(d,j) + P_{down}(d,j)\leq P_{max}(d,j)
+
+.. math:: 
+  :name: device_down_derivative_bounds
+
+    min(P_{min}(d,j),0) \leq P_{down}(d,j)\leq 0
+
+
+.. math:: 
+  :name: device_up_derivative_bounds
+
+    0 \leq P_{up}(d,j)\leq max(P_{max}(d,j),0)
+
+
+.. math:: 
+  :name: ems_derivative_bounds
+
+    P^{ems}_{min}(j) \leq \sum_d P^{ems}(d,j) \leq P^{ems}_{max}(j)
+
+EMS Power
```

Comment:
The equalities below this header couple power flows within the system. The header could better reflect that.